src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -360,13 +344,7 @@ class TypelessStructSegment(BaseSegment):
             Delimited(
                 AnyNumberOf(
                     Sequence(
-                        OneOf(
```

Comment:
This was about 80% the same but wasn't allowing `TypelessStructSegment`. It seemed better to reuse the existing grammar than maintain this separately when it seems to be converging.